### PR TITLE
Parametric Preprocessor and EP-Launch error handling cleanup

### LINKED ISSUE #6573: Parametric Preprocessor leaves funny file and err file repeats last error message
Running the parametric preprocessor on Linux in an empty folder without giving any arguments leaves a bad error filename: 
`C:\Documents and Settings\Jason\My Documents\projects\EnergyPlusDev\ParametricInputs\dev 2009-08a\Debug\Cond-1-Parametric-LogicTestingIF.err`

### LINKED ISSUE #8251: Modify EP-Launch crash handling when eplusout.end is turned off
Issue overview
--------------
The new v9.4.0 OutputControl:Files object allows the user to disable the eplusout.end file. Both EP-Launch and EP-Launch3 assume that the simulation has crashed if there is no eplusout.end file and issues a message to that effect. So, EP-Launch needs to either mention this possibility in the warning message, or read the last line of the err output file to determine is there was a crash (eplusout.err is not optional, it is always produced).

See discussion in #8250.

### Details
Some additional details for this issue (if relevant):
 - Windows
 - Version of EnergyPlus 9.4.0

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Fixes #6573 - cleanup error messages in Parametric Preprocessor
   See https://github.com/NREL/EnergyPlus/issues/6573#issuecomment-697503289 for testfiles.
 - Fixes #8251 - Adjust EP-Launch to not rely on the presence of eplusout.end to determine run completion status.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Review by Myoldmopar — APPROVED** [review]
From purely a code review perspective these look great.  Better initialization and reinitialization of the error string, not using a local path, and using the correct loop index are all good fixes.  I can build and run to verify if needed, but maybe someone else already has...

**Comment by JasonGlazer** [comment]
@mjwitte and @Myoldmopar I noticed that no example files use the OUTPUTCONTROL:FILES input files.

**Comment by JasonGlazer** [comment]
@Myoldmopar  Please check the changes to the energyplus.py workflow file and see if it works for you.

**Review by mjwitte — COMMENTED** [review]
Tested EP-Launch classic with various error and crash conditions. All working as expected.

Tested EP-Launch 3.0.0 with the modified workflow. It works as expected with no .end file, but if all output files are turned off, a successful run still shows as failed:

`Workflow FAILED: Current workflow main function failed unexpectedly:[Errno 2] No such file or directory: 'C:\\EnergyPlus\\EnergyPlusV9-4-0-RC2\\ExampleFiles\\1ZoneUncontrolled-NoOutput.eso'`

**Comment by mjwitte** [comment]
@JasonGlazer Here are my test files. 
[ErrorOutputTestFiles.zip](https://github.com/NREL/EnergyPlus/files/5283111/ErrorOutputTestFiles.zip)

**Comment by Myoldmopar** [comment]
The check for a successful EnergyPlus run shouldn't depend on the .end or .err anymore.  EnergyPlus exits with proper exit codes under failure conditions, so [this block](https://github.com/NREL/EnergyPlus/blob/6d31ab7e7a28efd5b9e2eafaaa8d9b5af3d187fc/workflows/energyplus.py#L181-L186) is sufficient for checking a failed condition.  Now if there isn't an error or end file then we should just report that it is unknown errors or what-not.

**Review by mjwitte — COMMENTED** [review]
Retested EP-Launch 3. Works great now.
@Myoldmopar Handing off to you for final review/merge.

**Comment by Myoldmopar** [comment]
OK, I tested this out.  I needed to set up an updated dev environment for EP-Launch3 since I haven't worked on it in a while.  When I did, I found the wxpython install to be much easier now, but I ended up with a brand new version of wxPython, and a few of the APIs had broken.  After working through a couple issues, I was able to run the E+ workflow.  I just ran it on 1ZoneUncontrolled and it worked, but there were errors in the workflow.  A variable had been used uninitialized in Python.  That happened a couple times and I tweaked the variables in the energyplus workflow here to account for where they were declared, along with a couple whitespace (tabs) clean up.  After that was done I ran this workflow on the newly added OutputControl file over in @mbadams5 branch.  I ran it by default (with end file) and it ran fine.  I then disabled the end file, cleaned out results, and re-ran it with success again.

I will plan on merging this as long as CI is fine with it -- it should be.  I will walk through the last couple tweaks I made to the workflow, though I will note there are other warnings lingering in here that my Python IDE is telling me about.  We should plan to clean all them up soon.

File: workflows/energyplus.py [hunk 1]
```diff
@@ -191,21 +191,24 @@ def run_energyplus(self, isIP, run_directory, file_name, args):
...
                 eplusouteso_path = os.path.join(run_directory, 'eplusout.eso')
                 if os.path.exists(eso_path):
-                		shutil.copy(eso_path, eplusouteso_path)
+                    shutil.copy(eso_path, eplusouteso_path)
                 # *.mtr back to eplusout.mtr
                 mtr_path = os.path.join(run_directory, file_name_no_ext + '.mtr')
                 eplusoutmtr_path = os.path.join(run_directory, 'eplusout.mtr')
                 if os.path.exists(mtr_path):
-                		shutil.copy(mtr_path, eplusoutmtr_path)
+                    shutil.copy(mtr_path, eplusoutmtr_path)
```
Comment (Myoldmopar) [inline workflows/energyplus.py:199]: Tabs -> Spaces

File: workflows/energyplus.py [hunk 2]
```diff
@@ -191,21 +191,24 @@ def run_energyplus(self, isIP, run_directory, file_name, args):
...
-                        convertESOMTR_binary = os.path.join(energyplus_root_folder, 'PostProcess', 'convertESOMTRpgm', 'convertESOMTR.exe')
+                        convertESOMTR_binary = os.path.join(energyplus_root_folder, 'PostProcess', 'convertESOMTRpgm',
+                                                            'convertESOMTR.exe')
                     else:
-                        convertESOMTR_binary = os.path.join(energyplus_root_folder, 'PostProcess', 'convertESOMTRpgm', 'convertESOMTR')
+                        convertESOMTR_binary = os.path.join(energyplus_root_folder, 'PostProcess', 'convertESOMTRpgm',
+                                                            'convertESOMTR')
                     if os.path.exists(convertESOMTR_binary):
-                        converttxt_orig_path = os.path.join(energyplus_root_folder, 'PostProcess', 'convertESOMTRpgm', 'convert.txt')
+                        converttxt_orig_path = os.path.join(energyplus_root_folder, 'PostProcess', 'convertESOMTRpgm',
```
Comment (Myoldmopar) [inline workflows/energyplus.py:210]: The line length was causing a warning.

File: workflows/energyplus.py [hunk 3]
```diff
@@ -230,7 +233,7 @@ def run_energyplus(self, isIP, run_directory, file_name, args):
                         if os.path.exists(ipmtr_path):
                             shutil.copy(ipmtr_path, mtr_path)
                             os.replace(ipmtr_path, eplusoutmtr_path)
-                    os.remove(converttxt_run_path)
+                        os.remove(converttxt_run_path)
```
Comment (Myoldmopar) [inline workflows/energyplus.py:236]: If you look above, `converttxt_run_path` was only declared inside the `if os.path.exists(convertESOMTR_binary)` block, and this delete call was outside of that block, so this was quietly failing.

File: workflows/energyplus.py [hunk 4]
```diff
@@ -304,19 +307,21 @@ def run_energyplus(self, isIP, run_directory, file_name, args):
                     if os.path.exists(readvars_audit_path):
                         os.replace(readvars_audit_path, rv_audit_path)
 
-                # clean up
-                if os.path.exists(temp_rvi_path):
-                    os.remove(temp_rvi_path)
-                if os.path.exists(temp_mvi_path):
-                    os.remove(temp_mvi_path)
+                    # clean up things inside this IF block
```
Comment (Myoldmopar) [inline workflows/energyplus.py:310]: Similarly, all these variables were only declared inside `if os.path.exists(readvarseso_binary)` but they were attempting to be deleted from outside the block, so being used uninitialized.

**Review by Myoldmopar — APPROVED** [review]
I'm happy with this, and it should merge today.

**Comment by Myoldmopar** [comment]
@mjwitte or @JasonGlazer if you could pull this branch and just verify my simple changes to the workflow didn't break anything that would be pretty great.  I would love to get the workflows unit tested so that I feel more free to refactor some of the code later.

**Comment by Myoldmopar** [comment]
I took another look at the changes, I think that they are quite reasonable.  The variables are now only used in the part of code they are defined/accessible.  I'm going to merge this.  If someone happens to try it out tomorrow, that's fine, but if not, I really feel OK about the minor change.

**Pull request merged**